### PR TITLE
STN-116 : Search Bar Polish

### PR DESCRIPTION
**Pull request by heatherdesigns**
# [STN-116](https://sparkbox.atlassian.net/browse/STN-116) READY FOR REVIEW

## Summary
Ensure the search bar looks and works as desired in the colorful theme.

Designs: 
[Figma desktop](https://www.figma.com/file/toNZnFLadtpNpG5dPzNGWz/⚙%EF%B8%8F-Colorful-Theme---Working?node-id=3603%3A282)
[Figma mobile](https://www.figma.com/file/toNZnFLadtpNpG5dPzNGWz/⚙%EF%B8%8F-Colorful-Theme-Working?node-id=3547%3A832)

## Need Review By (Date)
asap

## Urgency
high 🔥

## Steps to Test
1. In the CLI run `npm run test` to verify all tests still pass.
2. Go to http://economics.suhumsci.loc and verify the search input chisel meets the following acceptance criteria:
    - [x] Ensure that the search bar looks and works as expected after pairing with a designer and getting approval from the H&S team. I paired with Merani on design.
    - [x] Ensure that things work across all supported browsers.
    - [x] Ensure that things meet our accessibility guidelines

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss [hunk 1]
```diff
@@ -12,35 +11,39 @@
...
+    background-image: svg('<svg viewBox="0 0 25 26"><path fill-rule="evenodd" clip-rule="evenodd" d="M21.4737 10.2016C21.4737 14.0385 18.3632 17.149 14.5263 17.149C10.6894 17.149 7.57895 14.0385 7.57895 10.2016C7.57895 6.36466 10.6894 3.25422 14.5263 3.25422C18.3632 3.25422 21.4737 6.36466 21.4737 10.2016ZM24 10.2016C24 15.4338 19.7585 19.6753 14.5263 19.6753C12.5029 19.6753 10.6277 19.0409 9.08862 17.9603L2.11994 24.3929C1.60732 24.8661 0.808175 24.8341 0.334991 24.3215C-0.138192 23.8089 -0.106226 23.0098 0.406389 22.5366L7.22679 16.2408C5.86882 14.6013 5.05264 12.4968 5.05264 10.2016C5.05264 4.96942 9.29415 0.727905 14.5263 0.727905C19.7585 0.727905 24 4.96942 24 10.2016Z" stroke="#{$su-color-driftwood}" stroke-width="0.5" stroke-linecap="square" fill="#{$su-color-driftwood}" /></svg>');
```
Comment (bryanbraun) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss]: Great use of the SCSS variables here!

File: docroot/themes/humsci/humsci_basic/templates/page.html.twig [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
 
-  <section class="hb-page-width">
+  <section class="hb-page-width hb-position-relative">
     {% if page.header %}
       {{ page.header }}
     {% endif %}
 
     {% if page.search %}
-    <div class="su-site-search">
+    <div class="su-site-search" role="search">
```
Comment (bryanbraun) [inline docroot/themes/humsci/humsci_basic/templates/page.html.twig:12]: Great addition!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss [hunk 3]
```diff
@@ -12,35 +11,39 @@
...
     background-repeat: no-repeat;
     background-position: center center;
     border-width: 0;
-    opacity: 0.6;
     position: absolute;
     top: 0;
-    right: 1.2rem;
-    width: 24px;
-    height: 25px;
+    right: 14px;
```
Comment (kaseybon) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss]: We should use our `hb-calculate-rems()` function here.

**Review by kaseybon — CHANGES_REQUESTED** [review]
The search button alignment is off when logged out of Drupal and we've also lost the focus and hover states for the search button.

Screen Shot 2020-02-26 at 4 06 03 PM

---

There are also some issues with the mobile nav icon not lining up and the mobile nav overlaying the search input. I don't think it's within the scope of this card but can you please make a note in STN-210 that these will need to be resolved?

edit: this is affecting dropdowns in desktop views too.

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr499-nuinpeaifcojjeikvruwxzkyejynzz52.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e58067f8c1e65164ef749e8

File: docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss [hunk 4]
```diff
@@ -1,3 +1,46 @@
...
+  .su-lockup {
+    z-index: $hb-z-index-lockup;
+  }
+
+  .su-site-search {
+    display: block;
+    padding-top: hb-calculate-rems(12px);
+    padding-bottom: hb-calculate-rems(28px);
+    max-width: hb-calculate-rems(265px);
+    z-index: $hb-z-index-2;
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss]: you could  create a named variable for the z index like you did on line 3 above. for instance, `search-bar-z-index` would be good!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss [hunk 5]
```diff
@@ -1,3 +1,46 @@
...
+.su-masthead {
+  .su-lockup {
+    z-index: $hb-z-index-lockup;
+  }
+
+  .su-site-search {
+    display: block;
+    padding-top: hb-calculate-rems(12px);
+    padding-bottom: hb-calculate-rems(28px);
+    max-width: hb-calculate-rems(265px);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss]: let's put this in settigs/variables.general.scss and make a named variable for 256px

File: docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss [hunk 6]
```diff
@@ -1,3 +1,46 @@
...
+  }
+
+  .su-main-nav {
+    padding-bottom: hb-calculate-rems(14px);
+
+    .su-main-nav__toggle {
+      position: absolute;
+      right: 0;
+      bottom: 80%;
+      transform: translate(0, -80%); // Center the toggle on nav styles implementation. It is not centered while using Decanter's default nav styles because they overlay ontop of the lockup which makes the content inaccessible.
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss:21]: great comment!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss [hunk 7]
```diff
@@ -1,3 +1,46 @@
...
+
+      @include grid-media-min('sm') {
+        bottom: 60%;
+        transform: translate(0, -60%);
+      }
+    }
+
+    // The positioning of the nav set here will change as Decanter overlays the mobile nav on top of other content and our designs have the mobile nav inline with the content.
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss]: can you break this comment into two lines so it is a more readable line length? thanks!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss [hunk 8]
```diff
@@ -10,37 +9,46 @@
...
   }
 
+  .form-actions {
+    position: relative;
+  }
+
   input {
     @include font-smoothing;
-    @include padding(0.6rem 2rem);
+    padding: hb-calculate-rems(12px) hb-calculate-rems(36px) hb-calculate-rems(10px) hb-calculate-rems(14px);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:18]: 👍

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss [hunk 9]
```diff
@@ -10,37 +9,46 @@
...
 
     display: inline-block;
-    height: 4rem;
+    height: hb-calculate-rems(38px);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:21]: :good catch!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss [hunk 10]
```diff
@@ -10,37 +9,46 @@
...
+    border-radius: hb-calculate-rems(4px);
+    font-size: hb-calculate-rems(16px);
 
     &::placeholder {
-      color: $su-color-black;
+      color: $hb-color--black;
       opacity: 1; // For Firefox
+
+      @include hb-colorful {
+        color: darken($su-color-driftwood, 35%);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss]: for the driftwood color, you dont have to use the hb-colorful theme mixin if you dont want!

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr499-nuinpeaifcojjeikvruwxzkyejynzz52.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e58067f8c1e65164ef749e8

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss [hunk 11]
```diff
@@ -10,37 +9,46 @@
...
     position: absolute;
-    top: 0;
-    right: 1.2rem;
-    width: 24px;
-    height: 25px;
+    bottom: 0;
+    right: hb-calculate-rems(14px);
+    width: hb-calculate-rems(18px);
+    height: hb-calculate-rems(15px);
+    transform: scaleX(-1);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:46]: just curious, what gets scaled?

File: docroot/themes/humsci/humsci_basic/templates/page.html.twig (hunk 2 above)
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/templates/page.html.twig:12]: 👍

**Review by silverli — CHANGES_REQUESTED** [review]
looks pretty good! just a couple small requests :)

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss (hunk 11 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:46]: Nope, it just flips it. It's a hack I learned on a previous project. 😄

File: docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss (hunk 5 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss]: I placed it in `_variables/forms` instead. Let me know if that's ok.

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss (hunk 10 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss]: Good to know! So driftwood is going to be consistent throughout all the themes then?

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr499-nuinpeaifcojjeikvruwxzkyejynzz52.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e58067f8c1e65164ef749e8

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss (hunk 10 above)
Comment (silverli) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss]: yes! it's a part of the stanford identity colors

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss [hunk 12]
```diff
@@ -10,37 +9,42 @@
...
+    height: hb-calculate-rems(38px);
     max-width: 100%;
-    border-radius: 3rem;
-    font-size: 1.6rem;
+    border-radius: hb-calculate-rems(4px);
+    font-size: hb-calculate-rems(16px);
 
     &::placeholder {
-      color: $su-color-black;
+      color: darken($su-color-driftwood, 35%);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:27]: 👍

File: docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss (hunk 11 above)
Comment (silverli) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:46]: 💥

File: docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.forms.scss [hunk 13]
```diff
@@ -3,6 +3,7 @@ $hb-input-min-height: 38px;
 
 $hb-input-border-radius: 4px;
 $hb-input-border: 1px solid $su-color-driftwood;
+$hb-input-max-width: 265px;
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.forms.scss:6]: 👍

File: docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.zindex.scss [hunk 14]
```diff
@@ -19,4 +19,5 @@ $hb-z-index-12: 1200;
 // $hb-z-index-modal: $hb-z-index-1;
 $hb-z-index-main-menu: $hb-z-index-1;
 $hb-z-index-lockup: $hb-z-index-5; // keeps the lockup from laying on top of the admin menu
+$hb-z-index-search: $hb-z-index-5; // keeps the search input from being hidden by the navigation
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.zindex.scss:22]: nice!

**Review by silverli — COMMENTED** [review]
some really weird stuff happening with ie 11. let me know if you can replicate. we can always make a bug card if it feels out of scope


Screen Shot 2020-02-28 at 11 43 58 AM
windows-10-ie-11

**Pull request merged**